File: templates/apimethods.py
#! /usr/bin/python3
# -*- coding: utf-8 -*-

import sys, shutil, os, re, codecs, pprint
import requests, json, urllib3

# ----------------------------------------
def getJSON():
# ----------------------------------------
    with codecs.open("data.json", "r", encoding='utf-8') as dataFile:
        data = dataFile.read()
    return data


# ----------------------------------------
def prettifyJSON(sBody):
# ----------------------------------------

    oJSON = json.loads(sBody)
    sFormatJSON = json.dumps(oJSON, ensure_ascii = False, indent=4)
    return sFormatJSON


# ----------------------------------------
def RunTestGET(sURL):
# ----------------------------------------
    headers = {'content-type': 'application/json'}
    resp   = requests.get(sURL, headers=headers, verify=False)
    body   = prettifyJSON(resp.content)
    resp.connection.close()
    print(prettifyJSON(body))



# ----------------------------------------
def RunTestPOST(sURL):
# ----------------------------------------
    sData  = getJSON()
    headers = {'content-type': 'application/json'}
    resp   = requests.post(sURL, data=sData, headers=headers, verify=False)
    body   = prettifyJSON(resp.content)
    resp.connection.close()
    print(resp.status_code)
    print(body)


# ----------------------------------------
def RunTestPUT(sURL):
# ----------------------------------------
    sData  = getJSON()
    headers = {'content-type': 'application/json'}
    resp   = requests.put(sURL, data=sData, headers=headers, verify=False)
    body   = prettifyJSON(resp.content)
    resp.connection.close()
    print(resp.status_code)
    print(body)


# ----------------------------------------
def RunTestPATCH(sURL):
# ----------------------------------------
    sData  = getJSON()
    headers = {'content-type': 'application/json'}
    resp   = requests.patch(sURL, data=sData, headers=headers, verify=False)
    body   = prettifyJSON(resp.content)
    resp.connection.close()
    print(resp.status_code)
    print(body)


# ----------------------------------------
def RunTestDELETE(sURL):
# ----------------------------------------
    headers = {'content-type': 'application/json'}
    resp   = requests.delete(sURL, headers=headers, verify=False)
    body   = prettifyJSON(resp.content)
    resp.connection.close()
    print(resp.status_code)
    print(body)

# ----------------------------------------
def main():
# ----------------------------------------

    sMethod = sys.argv[1]
    sURI    = sys.argv[2]

    urllib3.disable_warnings()
    sURL = "[URL]" + sURI
    if (re.search("get|GET",sMethod)):
        print("GET Method")
        RunTestGET(sURL)
    elif (re.search("post|POST",sMethod)):
        print("POST Method")
        RunTestPOST(sURL)
    elif (re.search("put|PUT",sMethod)):
        print("PUT Method")
        RunTestPUT(sURL)
    elif (re.search("patch|PATCH",sMethod)):
        print("PATCH Method")
        RunTestPATCH(sURL)
    elif (re.search("delete|DELETE",sMethod)):
        print("DELETE Method")
        RunTestDELETE(sURL)
    else:
        print(":: ERROR :: UNKNOWN METHOD")

if __name__ == '__main__':
    main()

File: templates/apimethods.pl
use strict;
use Encode;
use File::Slurp;
use LWP::UserAgent;
use LWP::Simple;
use HTTP::Request;
use Data::Dumper;

# ------
# __main__
# ------
{
  my $sProd   = $ARGV[0];
  my $sMethod = $ARGV[1];
  my $sURI    = $ARGV[2];

  getResponse($sProd,$sMethod,$sURI);
}

# --------------------------------------------------------------------
sub getResponse($$$)
# --------------------------------------------------------------------
{
  my $hProd = {
 .  '<product_key>' => '<product_value>'
  };

  my $header = ['Content-Type' => 'application/json; charset=UTF-8'];
  my ($sProd,$sMethod,$sURI) = @_;
  my $sURL = $hProd->{$sProd} . $sURI;
  $sMethod = uc($sMethod);
  my $useragent = LWP::UserAgent->new;

  if ($sMethod=~/(GET|DELETE)/)
  {
    my $req = HTTP::Request->new($sMethod, $sURL, $header);
    my $res = $useragent->request($req);
    my $sResponse = $res->as_string;
    $sResponse = decode("utf8",$sResponse);

    open(OUT,">:utf8","out.json") or die "Cannot open out.json";
    print OUT $sResponse;
    close OUT;

    print $sResponse;
  }
  elsif ($sMethod=~/(POST|PUT|PATCH)/)
  {
    my $sContent = read_file("data.json");
    my $req = HTTP::Request->new($sMethod, $sURL, $header, $sContent);
    my $res = $useragent->request($req);
    my $sResponse = $res->as_string;
    $sResponse = decode("utf8",$sResponse);

    open(OUT,">:utf8","out.json") or die "Cannot open out.json";
    print OUT $sResponse;
    close OUT;

    print $sResponse;
  }
  else
  {
    print "ERROR:: Unknown method."
  }
}
